be/src/exec/spill/dir_manager.cpp

```diff
@@ -76,7 +77,14 @@ Status DirManager::init() {
             }
             return true;
         }));
-        _dirs.emplace_back(std::make_shared<Dir>(spill_dir_path, std::move(fs)));
+        ASSIGN_OR_RETURN(auto space_info, FileSystem::Default()->space(spill_dir_path));
+        int64_t max_dir_size = space_info.capacity;
+        for (const auto& storage_path : storage_paths) {
+            if (is_same_disk(spill_dir_path, storage_path)) {
+                max_dir_size = max_dir_size * max_dir_size_ratio;
+            }
+        }
```

Comment:
what will happen if we have multi storage_paths under the same disk?
suppose there are 2 storage_paths  and 1 spill_dir_path, they are all on the same disk,
according to the current calculation logic,
max_dir_size will be `disk_capacity* max_dir_size_ratio * max_dir_size_ratio`?

Reply:
how about breaking here? should we consider multi spill dir path on the same disk?

Reply:
> how about breaking here? should we consider multi spill dir path on the same disk? @silverbullet233

breaking here is ok. I think it doesn't make sense to configure multi dirs under the same disk.
but if it really happens, we can:
1. add a check in init stage and return an error to user
2. accept user's configuration and make sure all dirs under the same disk share the capacity limit. e.g. there are 2 spill_dir_path under same disk, each one can use `disk_capacity * max_dir_size_ratio / 2`
3. accept user's configuration but only use the first dirs in the same disk

I think all of above are ok, which one do you prefer? @caneGuy

Reply:
actually, i prefer return an error to user, i think one directory under the same disk is enough @silverbullet233

Reply:
> actually, i prefer return an error to user, i think one directory under the same disk is enough @silverbullet233

OK~